tests/messages.cpp

```diff
@@ -856,3 +899,32 @@ TEST(RoomEvents, ThreadedMessage)
 
     EXPECT_EQ(data.dump(), json(event).dump());
 }
+
+TEST(RoomEvents, InvalidMessage)
+{
+    json data = R"({
+          "origin_server_ts": 1510489356530,
+          "sender": "@nheko_test:matrix.org",
+          "event_id": "$15104893562785758wEgEU:matrix.org",
+          "unsigned": {
+            "age": 2225,
+            "transaction_id": "m1510489356267.2"
+          },
+          "content": {
+            "body": "I don't have a msgtype :(",
+            "m.relates_to": {
+              "m.in_reply_to": {
+                "event_id": "$6GKhAfJOcwNd69lgSizdcTob8z2pWQgBOZPrnsWMA1E"
+              }
+            }
+          },
+          "type": "m.room.message",
+          "room_id": "!lfoDRlNFWlvOnvkBwQ:matrix.org"
+    })"_json;
+
+    auto messageType         = getMessageType(data["content"]);
+    RoomEvent<Unknown> event = data.get<RoomEvent<Unknown>>();
```

Comment:
Hm, this doesn't really test the `parse_timeline_events` function. And when you fix that, you could also add a test for an int msgtype :3

Reply:
As in the body would contain `"msgtype": 1`?

Reply:
Yep

Reply:
Basically I want some of the other branches to be covered.

Reply:
I assume we would drop int msgtypes because they are not spec-compliant?

Reply:
How does it look now?